src/resolve.ts

```diff
@@ -0,0 +1,38 @@
+import {NonspatialScaleChannel, SCALE_CHANNELS, SPATIAL_SCALE_CHANNELS, SpatialScaleChannel, UNIT_SCALE_CHANNELS} from './channel';
+import * as log from './log';
+import {contains} from './util';
+
+export type ResolveMode = 'independent' | 'shared';
+
+export interface SpatialResolve {
+  scale?: ResolveMode;
+  axis?: ResolveMode;
+}
+
+export interface NonspatialResolve {
+  scale?: ResolveMode;
+  legend?: ResolveMode;
+}
+
+export type Resolve = SpatialResolve | NonspatialResolve;
+
+export type ResolveMapping = {[P in NonspatialScaleChannel]?: NonspatialResolve} | {[P in SpatialScaleChannel]?: SpatialResolve};
```

Comment:
(After changing to `&`, it would be great to check if we still need to do `sed` -- maybe not anymore.)